src/ovro_lwa_portal/io.py

```
@@ -117,70 +194,215 @@ def _detect_source_type(source: str | Path) -> tuple[str, str]:
+                raise ImportError(msg) from e
+
+            # Create UPath object - works for both local and remote paths
+            # Pass storage_options so credentials reach the filesystem backend
+            if storage_options:
+                store_path = UPath(normalized_source, **storage_options)
+            else:
+                store_path = UPath(normalized_source)
+
+            # Explicit local existence check
+            if store_path.protocol in ("", "file") and not store_path.exists():
+                raise FileNotFoundError(f"Local path does not exist: {store_path}")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)